File: src/api/fmpService.js
const fs = require('fs/promises');
const path = require('path');
const config = require('../../config');

let cacheData = null;

async function loadData() {
  if (!cachedData) {
    const filePath = path.resolve(__dirname, '../../', config.dataFile);
    const text = await fs.readFile(filePath, 'utf-8');
    cachedData = JSON.parse(text);
  }
  return cachedData;
}

function cleanData(raw) {
  if (Array.isArray(raw) && raw.length > 0 && raw[0].date === 'Date') {
    return raw.slice(1);
  }
  return raw;
}

async function getMarketHistoricalData(period, symbol) {
  const data = await loadData();
  const raw = data[symbol];
  if (!raw) {
    throw new Error(`Symbol ${symbol} not found in local data`);
  }
  return { data: cleanData(raw) };
}


// 此为调用fmp接口
// async function getMarketHistoricalData(period, symbol) {
//   const response = await fetch(`https://backend.ymyc.ai/api/market-historical-data/${period}/${symbol}`);
//   if (!response.ok) {
//     throw new Error('Failed to get stock price');
//   }
//   const jsonData = await response.json();
//   return jsonData;
// }

module.exports = {
  getMarketHistoricalData,
};

File: src/api/fmp_service.py
import json
from pathlib import Path
import urllib.request


BASE_URL = 'https://backend.ymyc.ai/api/market-historical-data'


def get_market_historical_data(period: int, symbol: str):
    """Fetch historical market data from remote service."""
    url = f"{BASE_URL}/{period}/{symbol}"
    try:
        with urllib.request.urlopen(url) as resp:
            if resp.status != 200:
                raise RuntimeError('Failed to get stock price')
            data = json.load(resp)
    except Exception:
        # Fallback: try local JSON file if network is unavailable
        local_file = Path(f"{symbol}.json")
        if local_file.exists():
            with local_file.open('r', encoding='utf-8') as f:
                data = json.load(f)
        else:
            raise
    return data
